src/audio/multiband_drc/multiband_drc.c

```diff
@@ -370,14 +379,21 @@ static int multiband_drc_init(struct processing_module *mod)
 	ret = comp_init_data_blob(cd->model_handler, bs, cfg->data);
 	if (ret < 0) {
 		comp_err(dev, "multiband_drc_init(): comp_init_data_blob() failed.");
-		goto cd_fail;
+		goto cd_model_fail;
 	}
 	multiband_drc_reset_state(&cd->state);
 
+	/* Initialize to enabled is a workaround for IPC4 kernel version 6.6 and
+	 * before where the processing is never enabled via switch control. New
+	 * kernel sends the IPC4 switch control and sets this to desired state
+	 * before prepare.
+	 */
+	multiband_drc_process_enable(&cd->process_enabled);
 	return 0;
 
-cd_fail:
+cd_model_fail:
```

Comment:
again this isn't needed. cd is initialized to all NULL and the function is smart enough to check to see if NULL is passed in. So this split logic just adds more paths for no reason.